homeassistant/components/homekit_controller/__init__.py

```diff
@@ -148,11 +156,25 @@ def accessory_setup(self):
                 service_info = {'serial': serial,
                                 'aid': aid,
                                 'iid': service['iid']}
-                devtype = homekit.ServicesTypes.get_short(service['type'])
-                _LOGGER.debug("Found %s", devtype)
-                component = HOMEKIT_ACCESSORY_DISPATCH.get(devtype, None)
+                stype = service['type']
+                kservice = HOMEKIT_KNOWN_SERVICE_TYPES.get(stype, None)
+                # If service type ID is known, it is dispatched to platform
+                if kservice is not None:
+                    _LOGGER.debug("Mapped service: %s type: %s platform: %s",
+                                  stype,
+                                  kservice['type'],
+                                  kservice['platform'])
+                    component = kservice['type']
+                    devdomain = kservice['platform']
```

Comment:
I don't think we want to do this. A component should only set up its own platforms.

Reply:
This would need approval by core or in architecture issue. But I don't think it will be approved.

Reply:
This should not be a standalone platform but instead be part of homekit sensor.

Reply:
This is specific implementation for Koogeek service, and not the generic implementation of HomeKit Service that would be handled by HomeKit Controller Sensor. I can do it in HomeKit Controller sensor but it is specific only to Koogeek. After a while the HomeKit Controller Sensor/Switch ... would get too big with all the different vendor's service implementations. I am not sure it would be correct. Do you think it would be better in HomeKit Controller Sensor even if it is vendor specific ?

I tried asking on Discord without luck so I implemented it somehow to get feedback.

Reply:
it's fine if the homekit sensor grows too big, we can always split it up in multiple files (`sensor/homekit/koogeek.py` etc)

Reply:
CC @cdce8p

Reply:
CC: @mjg59

Reply:
Oki thanks for clarification, will do it that way then :)
What do you think about the "dispatching" (mapping) of HomeKit services to Home Assistant ? Do you think that it should be in HomeKit Controller like now or do you think it would be better in switch/sensor ... ?